cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -264,6 +271,13 @@ public String call() throws Exception {
                   LOG.debug("Got file {} with start time {}", file, Bytes.toLong(colName));
                 }
 
+                if (Strings.isNullOrEmpty(namespacedLogDir)) {
+                  LOG.warn("Delete metadata for {} namespace", namespaceId.getEntityName());
+                  table.delete(rowKey, colName);
+                  deletedColumns++;
+                  continue;
+                }
+
                 Location fileLocation = impersonator.doAs(namespaceId, new Callable<Location>() {
```

Comment:
Can we assume that this will not fail with NamespaceNotFound? Because of the previous doAs() above? I guess it can only happen if the namespace is deleted while this method is running. I think if that happens, we want to ignore this entry, and not fail the entire transaction.